crates/common/chain/lean/src/lean_chain.rs

```diff
@@ -287,43 +302,47 @@ impl LeanChain {
         let add_attestations_timer =
             start_timer_vec(&PROPOSE_BLOCK_TIME, &["add_valid_attestations_to_block"]);
         loop {
-            state.process_attestations(&new_block.message.body.attestations)?;
-            let new_attestations_to_add = latest_known_attestation_provider
-                .get_all_attestations()?
-                .into_iter()
-                .filter_map(|(_, attestation)| {
-                    (attestation.message.source() == state.latest_justified
-                        && !new_block.message.body.attestations.contains(&attestation))
-                    .then_some(attestation)
-                })
-                .collect::<Vec<_>>();
+            state.process_attestations(&new_block.body.attestations)?;
+            let mut new_attestations_to_add = Vec::new();
+            let mut new_signatures_to_add = Vec::new();
+
+            for (_, attestation) in latest_known_attestation_provider.get_all_attestations()? {
+                if attestation.message.source() == state.latest_justified
+                    && !new_block.body.attestations.contains(&attestation.message)
+                {
+                    new_attestations_to_add.push(attestation.message);
+                    new_signatures_to_add.push(attestation.signature);
+                }
+            }
```

Comment:
```suggestion
            for signed_attestation in latest_known_attestation_provider
                .get_all_attestations()?
                .values()
            {
                if signed_attestation.message.source() == state.latest_justified
                    && !new_block
                        .body
                        .attestations
                        .contains(&signed_attestation.message)
                {
                    new_attestations_to_add.push(signed_attestation.message.clone());
                    new_signatures_to_add.push(signed_attestation.signature);
                }
            }
```

- Using `values()`
- Rename `attestation` -> `signed_attestation`